### PR TITLE
docs: use timeout in create dataset sample

### LINKED ISSUE #8: BigQuery: Document the use of the timeout parameter in samples
After adding the new `timeout` parameter to various public methods (#9987), we should demonstrate its usage in the code samples.

Users should be aware of this new feature, and should probably use it by default to avoid sporadic weird issues related to a method "getting stuck" at the transport layer.

### PR DESCRIPTION
**Pull request by plamut**
Fixes #8.

### PR checklist
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by plamut** [comment]
From the [issue thread](https://github.com/googleapis/python-bigquery/issues/8#issuecomment-566785126):
> Makes sense to me. I think that once we have a timeout parameter on all relevant methods, we can surface an example in an API best practices doc such as https://cloud.google.com/bigquery/docs/api-performance

@tswast What's the repo for changing that, is it public? If not, somebody else should update those docs and mention timeouts among recommended good practices.

**Comment by tswast** [comment]
> @tswast What's the repo for changing that, is it public? If not, somebody else should update those docs and mention timeouts among recommended good practices.

It's in `google3` (private). @shollyman or one of the other BQ DevRel folks should be able to take care of that.

File: samples/create_dataset.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ def create_dataset(dataset_id):
     # TODO(developer): Specify the geographic location where the dataset should reside.
     dataset.location = "US"
 
-    # Send the dataset to the API for creation.
+    # Send the dataset to the API for creation, with a timeout.
```
Comment (tswast) [inline samples/create_dataset.py]: Let's say "explicit timeout", since we have default values for timeouts now.

Comment (plamut) [inline samples/create_dataset.py]:
```suggestion
    # Send the dataset to the API for creation, with an explicit timeout.
```

**Pull request merged**